File: implementations/node/sdk/src/contracts/publication/pipeline-publication-response.ts
import { AI_SDK_SCHEMA_VERSIONS } from "../common/schema-versions.js";

export interface AiSdkPipelinePublicationResponse {
  readonly schemaVersion: typeof AI_SDK_SCHEMA_VERSIONS.pipelinePublicationResponse;
  readonly publicationRef: string;
  readonly publicationSha256: string;
  readonly pipelineName: string;
  readonly pipelineVersion: string;
}

File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/publication/pipeline_publication_response.py
from dataclasses import dataclass

from ...wire_model import AiSdkWireModel
from ..common.schema_versions import AiSdkSchemaVersions


@dataclass(frozen=True)
class AiSdkPipelinePublicationResponse(AiSdkWireModel):
    publication_ref: str
    publication_sha256: str
    pipeline_name: str
    pipeline_version: str
    schema_version: int = AiSdkSchemaVersions.PIPELINE_PUBLICATION_RESPONSE
